consensus/src/chained_bft/chained_bft_smr.rs

```diff
@@ -85,35 +95,50 @@ impl<T: Payload> ChainedBftSMR<T> {
     fn start_event_processing(
         executor: Handle,
         mut epoch_manager: EpochManager<T>,
-        mut pacemaker_timeout_sender_rx: channel::Receiver<Round>,
+        pacemaker_timeout_sender_rx: mpsc::Receiver<Round>,
         network_task: NetworkTask<T>,
         mut network_receivers: NetworkReceivers<T>,
-    ) {
-        let fut = async move {
+        shutdown: Arc<AtomicBool>,
+    ) -> thread::JoinHandle<()> {
+        executor.spawn(network_task.start());
+        thread::spawn(move || loop {
```

Comment:
here is the unfortunate situation... so let's think about this... we have a series of receivers. What we probably want to do is pass in a sender to each one of these services. That way we can actually merge these messages in a Rust friendly fashion. I think this shouldn't be too awful. Going to go back and request changes until we get this nailed down.